File: model/src/main/java/me/rolgalan/marvelw/model/Comic.java
package me.rolgalan.marvelw.model;

import java.util.List;

/**
 * Created by Roldán Galán on 20/11/2017.
 */

public interface Comic {

    Long getId();

    String getTitle();

    String getDescription();

    String getIsbn();

    String getImagePathThumbnail();

    List<SimpleExtraItem> getCharacters();

    List<SimpleExtraItem> getStories();

    List<SimpleExtraItem> getSeries();

}

File: model/src/main/java/me/rolgalan/marvelw/model/Comic.kt
package me.rolgalan.marvelw.model

/**
 * Created by Roldán Galán on 20/11/2017.
 */

interface Comic : Identifiable {

    val title: String

    val description: String

    val isbn: String

    val imagePathThumbnail: String

    val characters: List<SimpleExtraItem>

    val stories: List<SimpleExtraItem>

    val series: List<SimpleExtraItem>

}
